xbrowse_server/phenotips/views.py

```diff
@@ -1,90 +1,187 @@
+  '''test function to see if we can proxy phenotips'''  
+  current_user = request.user
+  
+  if request.GET.has_key('project'):
+    project_name=request.GET['project']  
+    #add project name to session for use in proxying
+    request.session['current_project_name']=project_name
+    
+    #also put current ext_id into session object
+    phenotips_proj_specific_admin__uname,phenotips_proj_specific_admin_pwd = get_uname_pwd_for_project(project_name)
+    ext_id=convert_internal_id_to_external_id(eid,
+                                            phenotips_proj_specific_admin__uname,
+                                            phenotips_proj_specific_admin_pwd)
+    request.session['current_ext_id']=ext_id
+    
+    #no check current auth level and add that to session too
+    project = get_object_or_404(Project, project_id=project_name)
+    if project.can_admin(request.user):
+        auth_level = 'admin'
+    elif project.can_edit(request.user):
+        auth_level = 'editor'
+    elif project.is_public:
+        auth_level = 'public'
+    elif project.can_view(request.user):
+        auth_level = 'viewer'
+    else:
+        raise Exception("Authx - cannot authenticate:fetch_phenotips_edit_page ")
+    request.session['current_auth_level']=auth_level
+    
+  else: 
+    #getting admin account to translate eid to id
+    project_name = request.session['current_project_name']
+    ext_id=request.session['current_ext_id']
+    auth_level = request.session['current_auth_level']
+    
+  #depending on auth level, pick either the full edit username or the view-only username for this
+  #project to fetch page
+  phenotips_uname,phenotips_pwd ='_','_'
+  if auth_level=='admin':
+    phenotips_uname,phenotips_pwd = get_uname_pwd_for_project(project_name)
+  else:
+    phenotips_uname,phenotips_pwd = get_generic_collaborator_uname_pwd_for_project(project_name)
+  url= settings.PHENOPTIPS_HOST_NAME+'/bin/'+ ext_id
+  if auth_level=='admin':
+    url= settings.PHENOPTIPS_HOST_NAME+'/bin/edit/data/'+ ext_id
+  if not request.GET.has_key('project'):
+    url += '?'
+    counter=0
+    for param,val in request.GET.iteritems():
+      url += param + '=' + val
+      if counter < len(request.GET)-1:
+        url += '&'
+      counter+=1
+  if type(ext_id) != dict:
+    #we are using the project name as the username and project name twice as the password
+    #for example if project name was foo, the username would be foo, password would be foofoo
+    result = do_authenticated_call_to_phenotips(url,phenotips_uname,phenotips_pwd)
+    response = __add_back_phenotips_headers_response(result)
+    return response
+  else:
+    logger.error('phenotips.views:'+ext_id['error'])
+    raise Http404    
 
 
+#do a GET as a proxy for Phenotips
+#exempting csrf here since phenotips doesn't have this support
+@csrf_exempt
+def proxy_get(request):
+  '''to act as a proxy for get requests '''
+  project_name = request.session['current_project_name']
```

Comment:
I think this relates to a Phenotips proxy quirk since calls come on different auth levels here. While this is a good idea, I think it breaks the proxy strategy. Let's chat about his in person soon and maybe figure out a compromise.